VitalTrackApp/src/screens/Analytics.js

```diff
@@ -1,31 +1,61 @@
-import React from 'react';
-import {View, Text, ScrollView, StyleSheet, Image} from 'react-native';
+import React, { useState, useEffect } from 'react';
+import {View, Text, ScrollView, StyleSheet, Alert} from 'react-native';
+import MoodGraph from '../components/MoodGraph';
+import axios from 'axios';
+import auth from '@react-native-firebase/auth';
+
 
 export default function AnalyticsScreen() {
+  const [tips, setTips] = useState(null);
+  const userId = auth().currentUser ? auth().currentUser.uid : null;
+
+  useEffect(() => {
+    const fetchTips = async () => {
+      if (!userId) return;
+
+      try {
+        const response = await axios.get(`http://10.0.2.2:5000/tips/${userId}`);
+        if (response.data) {
+          setTips(response.data);
+        } else {
+          Alert.alert('Error', 'No tips found');
+        }
+      } catch (error) {
+        console.log("Error fetching tips:", error);
+        Alert.alert('Error', error.message);
+      }
+    };
+
+    fetchTips();
+  }, [userId]);
+
+
   return(
     <ScrollView>
       <View style={styles.container}>
-        <Text style={styles.title}>Analytics Page</Text>
+        <View style={styles.header}>
+            <Text style={styles.headerText}>Analytics</Text>
+          </View>
         <Text style={styles.subtitle}>An overview of your wellbeing</Text>
 
         <View style={styles.section}>
             <Text style={styles.title}>Insights</Text>
-            <Text style={styles.subtitle}>You felt [less stressed] when you [did yoga] this week.</Text>
+            <Text style={styles.caption}>{tips ? tips.message : "Loading tips..."}</Text>
```

Comment:
when we render these tips or anything in the analytics page can we use the TextGeneratorEffect I created. You can simply Wrap it like this <TextGeneratorEffect text={tips}/>

Reply:
Okay, I changed it to use text generator effect